src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -1,119 +1,95 @@
-import React from 'react';
+import React, {forwardRef, useEffect, useState, useRef, useImperativeHandle, useCallback} from 'react';
 import {Dimensions} from 'react-native';
 import _ from 'underscore';
-import lodashGet from 'lodash/get';
 import * as Report from '../../../../libs/actions/Report';
 import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
 import PopoverWithMeasuredContent from '../../../../components/PopoverWithMeasuredContent';
 import BaseReportActionContextMenu from './BaseReportActionContextMenu';
 import ConfirmModal from '../../../../components/ConfirmModal';
-import CONST from '../../../../CONST';
 import * as ReportActionsUtils from '../../../../libs/ReportActionsUtils';
 import * as IOU from '../../../../libs/actions/IOU';
 
 const propTypes = {
     ...withLocalizePropTypes,
 };
 
-class PopoverReportActionContextMenu extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            reportID: '0',
-            reportAction: {},
-            selection: '',
-            reportActionDraftMessage: '',
-            isPopoverVisible: false,
-            isDeleteCommentConfirmModalVisible: false,
-            shouldSetModalVisibilityForDeleteConfirmation: true,
-            cursorRelativePosition: {
-                horizontal: 0,
-                vertical: 0,
-            },
-
-            // The horizontal and vertical position (relative to the screen) where the popover will display.
-            popoverAnchorPosition: {
-                horizontal: 0,
-                vertical: 0,
-            },
-            isArchivedRoom: false,
-            isChronosReport: false,
-            isPinnedChat: false,
-            isUnreadChat: false,
-        };
-        this.onPopoverShow = () => {};
-        this.onPopoverHide = () => {};
-        this.onPopoverHideActionCallback = () => {};
-        this.contextMenuAnchor = undefined;
-        this.showContextMenu = this.showContextMenu.bind(this);
-        this.hideContextMenu = this.hideContextMenu.bind(this);
-        this.measureContextMenuAnchorPosition = this.measureContextMenuAnchorPosition.bind(this);
-        this.confirmDeleteAndHideModal = this.confirmDeleteAndHideModal.bind(this);
-        this.hideDeleteModal = this.hideDeleteModal.bind(this);
-        this.showDeleteModal = this.showDeleteModal.bind(this);
-        this.runAndResetOnPopoverShow = this.runAndResetOnPopoverShow.bind(this);
-        this.runAndResetOnPopoverHide = this.runAndResetOnPopoverHide.bind(this);
-        this.getContextMenuMeasuredLocation = this.getContextMenuMeasuredLocation.bind(this);
-        this.isActiveReportAction = this.isActiveReportAction.bind(this);
-
-        this.dimensionsEventListener = null;
-
-        this.contentRef = React.createRef();
-        this.setContentRef = (ref) => {
-            this.contentRef.current = ref;
+const PopoverReportActionContextMenu = forwardRef((props, ref) => {
```

Comment:
The standard when we use `forwardRef` is to apply it at the end of the file, like

```
export default forwardRef((props, ref) => (
        <PopoverReportActionContextMenu
            // eslint-disable-next-line react/jsx-props-no-spreading
            {...props}
            innerRef={ref}
        />
    )),
    ```